### PR TITLE
Create smart_contracts.md

### PR DESCRIPTION
**Pull request by yvan-sraka**
[Rendered](https://github.com/massalabs/massa/blob/docs-wasm-vm/docs/smart_contracts.md)

### CODE REVIEW

File: docs/smart_contracts.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
+# WASM VM
+
+We want to use smart contracts encoded as WASM byte code.
+
+## State of the art of WASM implementation 
+
+### WebAssembly 101
+
+The resources on WebAssembly that I mentioned this morning were:
```
Comment (AureliaDolo) [inline docs/smart_contracts.md:9]: A little too specific

**Review by adrien-zinger — COMMENTED** [review]
we've gone so far since this first conversation. I'm afraid about forgetting this doc somewhere and that one day someone read that and think it's a real documentation :face_in_clouds:

File: docs/smart_contracts.md [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+https://github.com/adrien-zinger/wasm-simulation/issues/2
+
+### Solution A: use AssemblyScript little overhead
+
+- https://github.com/adrien-zinger/wasmer-as
+- https://github.com/near/assemblyscript-json
+
+### Solution B: use WasiEnv
+
+TODO
```
Comment (yvan-sraka) [inline docs/smart_contracts.md:123]: Reminder to myself to document this solution experimented here https://github.com/massalabs/wasm-simulation/tree/wasi-env

File: docs/smart_contracts.md [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+
+### Solution C: use WasmBindgen
+
+TODO
```
Comment (yvan-sraka) [inline docs/smart_contracts.md:127]: I was mistaking saying Solana SC engine was based somehow on https://github.com/rustwasm/wasm-pack

It smartly uses https://fr.wikipedia.org/wiki/BSD_Packet_Filter as shown in this example https://github.com/mvines/solana-bpf-program-template

**Comment by yvan-sraka** [comment]
@adrien-zinger did we just abandon this document or do we try to keep it up to date?

My plan was also to try to make an another doc to explain solutions explored to run high-level language on top of WASM runtime.

Do we put all these docs in some `hacking` or `internal` subfolder?

**Comment by yvan-sraka** [comment]
Moved here https://github.com/massalabs/sc-runtime/wiki/Smart-Contracts

**Pull request closed without merging**